src/wasm/wasm-validator.cpp

```diff
@@ -277,6 +289,7 @@ struct FunctionValidator : public WalkerPass<PostWalker<FunctionValidator>> {
     if (curr->is<Loop>()) {
       self->pushTask(visitPreLoop, currp);
     }
+    self->pushTask(visitExpression, currp);
```

Comment:
How about checking here if `getFunction()` and the profile is Poppy? Can then rename to `visitPoppyExpression`.